src/screens/Wallets/Receive/ReceiveConnect.tsx

```diff
@@ -26,15 +32,63 @@ const ReceiveConnect = ({
 	const { t } = useTranslation('wallet');
 	const { isSmallScreen } = useScreenSize();
 	const lightningBalance = useLightningBalance(true);
-	const { amount, jitOrder } = useSelector(receiveSelector);
+	const [feeEstimate, setFeeEstimate] = useState(0);
+	const [isLoading, setIsLoading] = useState(false);
+	const { amount } = useSelector(receiveSelector);
+	const invoice = useSelector(receiveSelector);
+	const blocktank = useSelector(blocktankInfoSelector);
 
-	const order = jitOrder!;
-	const payAmount = amount - order.feeSat;
-	const displayFee = useDisplayValues(order.feeSat);
+	const { maxChannelSizeSat } = blocktank.options;
 
-	const onContinue = (): void => {
+	const requestData: ICreateOrderRequest = {
+		lspBalanceSat: amount,
+		channelExpiryWeeks: DEFAULT_CHANNEL_DURATION,
+		options: {},
+	};
+
+	async function feeEstimation(): Promise<void> {
+		try {
+			const estimate = await estimateOrderFee(requestData);
+			if (estimate.isOk()) {
+				setFeeEstimate(estimate.value);
+			} else {
+				console.error('Error in estimate');
+			}
+		} catch (error) {
+			console.error('Error', error);
+		}
+	}
+
+	feeEstimation();
+
+	const payAmount = amount - feeEstimate;
+	const displayFee = useDisplayValues(feeEstimate);
+
+	const onContinue = async (): Promise<void> => {
+		setIsLoading(true);
+
+		const cJitEntryResponse = await createCJitEntry({
+			channelSizeSat: maxChannelSizeSat,
+			invoiceSat: invoice.amount,
+			invoiceDescription: invoice.message,
+			channelExpiryWeeks: DEFAULT_CHANNEL_DURATION,
+			couponCode: 'bitkit',
+		});
+		if (cJitEntryResponse.isErr()) {
+			setIsLoading(false);
+			console.log({ error: cJitEntryResponse.error.message });
+			showToast({
+				type: 'error',
+				title: t('receive_cjit_error'),
+				description: cJitEntryResponse.error.message,
+			});
+			return;
+		}
+		const order = cJitEntryResponse.value;
+		updateInvoice({ jitOrder: order });
 		addCJitEntry(order).then();
 		navigation.navigate('ReceiveQR');
+		setIsLoading(false);
```

Comment:
Only one `setIsLoading(false);` is needed here:

```suggestion
	const onContinue = async (): Promise<void> => {
		setIsLoading(true);

		const cJitEntryResponse = await createCJitEntry({
			channelSizeSat: maxChannelSizeSat,
			invoiceSat: invoice.amount,
			invoiceDescription: invoice.message,
			channelExpiryWeeks: DEFAULT_CHANNEL_DURATION,
			couponCode: 'bitkit',
		});
		if (cJitEntryResponse.isErr()) {
			showToast({
				type: 'error',
				title: t('receive_cjit_error'),
				description: cJitEntryResponse.error.message,
			});
			return;
		}
		const order = cJitEntryResponse.value;
		updateInvoice({ jitOrder: order });
		addCJitEntry(order).then();
		setIsLoading(false);
		navigation.navigate('ReceiveQR');
```